plugins/source/awspricing/resources/services.go

```diff
@@ -0,0 +1,320 @@
+package resources
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/cloudquery/cloudquery/plugins/source/awspricing/client"
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/transformers"
+)
+
+const ROOT_URL = "https://pricing.us-east-1.amazonaws.com"
+
+func Services() *schema.Table {
+	return &schema.Table{
+		Name:                "awspricing_services",
+		Resolver:            fetchServicesTable,
+		PreResourceResolver: getPricingFile,
+		Transform:           transformers.TransformWithStruct(&PricingFile{}, transformers.WithSkipFields("Products", "Terms")),
+		Relations: []*schema.Table{
+			products(),
+			terms(),
+		},
+	}
+}
+
+func fetchServicesTable(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan<- any) error {
+	c := meta.(*client.Client)
+	links, err := getPricingFileLinks(c)
+	if err != nil {
+		return err
+	}
+	res <- links
+	return nil
+}
+
+func validateOffersToSync(c *client.Client, offerCode string) bool {
+	// Can't use glob.Glob because currently it is in plugin-sdk/internal
+	for _, includeOfferCode := range c.OfferCodes {
+		// if glob.Glob(includeOfferCode, offerCode) {
+		if includeOfferCode == "*" || includeOfferCode == offerCode {
+			return true
+		}
+	}
+	return false
+}
+
+func getPricingFileLinks(c *client.Client) ([]string, error) {
+	resp, err := http.Get(ROOT_URL + "/offers/v1.0/aws/index.json")
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	offersFile := make(map[string]any, 0)
+
+	if err := json.NewDecoder(resp.Body).Decode(&offersFile); err != nil {
+		return nil, err
+	}
+	links := make([]string, 0)
+	for _, offer := range offersFile["offers"].(map[string]any) {
+		offer := offer.(map[string]any)
+		if !validateOffersToSync(c, offer["offerCode"].(string)) {
+			continue
+		}
+		full_url := ROOT_URL + offer["currentRegionIndexUrl"].(string)
+
+		regionalLinks, err := getRegionalPricingFileLinks(c, full_url)
+		if err != nil {
+			return nil, err
+		}
+		links = append(links, regionalLinks...)
+	}
+	return links, nil
+}
+
+func validateRegionsToSync(c *client.Client, regionCode string) bool {
+	// Can't use glob.Glob because currently it is in plugin-sdk/internal
+	for _, includeRegionCode := range c.RegionCodes {
+		// if glob.Glob(includeOfferCode, offerCode) {
+		if includeRegionCode == "*" || includeRegionCode == regionCode {
+			return true
+		}
+	}
+	return false
+}
+
+func getRegionalPricingFileLinks(c *client.Client, link string) ([]string, error) {
+	resp, err := http.Get(link)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	offerFiles := make(map[string]any, 0)
```

Comment:
I thought about creating a custom MarshalJSON but we can't do that because we have to transform the data. 


The we didn't do this the structure of file would mean it doesn't fit in a single row in any database. Some files are in excess of 250 MB